format_message.lua

```diff
@@ -35,15 +35,14 @@ beerchat.format_message = function(s, tab)
 		channel_name = minetest.colorize(color, channel_name)
 	end
 
-	local params = {
-		channel_name = channel_name,
-		channel_owner = owner,
-		channel_password = password,
-		from_player = tab.from_player,
-		to_player = tab.to_player,
-		message = colorize_target_name(tab.message, tab.to_player),
-		time = os.date("%X")
-	}
+	local params = table.copy(tab)
```

Comment:
Copying table for every possible message formatter is bad thing to do, this should be either removed or updated before pushing toward release.
For ACLs 2-step formatting would also work: first `beerchat.format_string(fmtstr, table)` and then format results with `beerchat.format_message(fmtstr, table)`.

Reply:
Removed table copy and switched to preformatting with direct beerchat.format_string call.